inference-engine/src/inference_engine/ie_blob_common.cpp

```diff
@@ -12,51 +12,49 @@
 
 namespace InferenceEngine {
 
-Blob::Ptr make_shared_blob(const Blob::Ptr& inputBlob, const ROI& roi) {
-    // reject compound blobs
-    if (inputBlob->is<CompoundBlob>()) {
-        THROW_IE_EXCEPTION << "Compound blobs do not support ROI";
-    }
-
-    size_t blkDimsH = roi.sizeY;
-    size_t blkDimsW = roi.sizeX;
-    size_t blkDimsC = inputBlob->getTensorDesc().getDims()[1];
+TensorDesc make_roi_tensor_desc(const TensorDesc& inputTensorDesc, const ROI& roi) {
```

Comment:
The function should take `roi.id` into account as well, since it has clear meaning now (index over batch dimension).

Reply:
Current logic is not being modified. 
This PR focuses on API and keeps the current behavior intact.